creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -176,13 +183,26 @@ class SnapbackSM {
         `stateMachineQueue Job Lock Extension Failed - ID ${jobId} - Error ${err}`
       )
     })
+
+    // Re-queue state machine job when the current job fails or succeeds
+    this.stateMachineQueue.on('global:error', (error) => {
+      this.logError(`stateMachineQueue Job Error - ${error}`)
+      this.log(`stateMachineQueue adding another job after error`)
+      this.stateMachineQueue.add({ startTime: Date.now() })
+    })
     this.stateMachineQueue.on('global:completed', (jobId, result) => {
       this.log(
         `stateMachineQueue Job Completed - ID ${jobId} - Result ${result}`
       )
+      this.log(
+        `stateMachineQueue adding another job after successful completion`
+      )
+      this.stateMachineQueue.add({ startTime: Date.now() })
     })
     this.stateMachineQueue.on('global:failed', (jobId, err) => {
       this.logError(`stateMachineQueue Job Failed - ID ${jobId} - Error ${err}`)
+      this.log(`stateMachineQueue adding another job after previous job failed`)
+      this.stateMachineQueue.add({ startTime: Date.now() })
```

Comment:
what's the difference between failed and error?

Reply:
the docs aren't very helpful, but it seems like failed is when a specific job fails and is not going to run anymore. error seems more like just a general error that's not necessarily tied to any one job. so if we're seeing duplicate jobs it could be that error shouldn't be re-queuing the job